### PR TITLE
Add contribution guidelines

### PR DESCRIPTION
**Pull request by flomnes**
About data members, I like the `m_` prefix, since it makes it very easy to identify member variables.

It's not our current convention, and it would take some effort to apply it to our current code base.

[source](https://github.com/organicmaps/organicmaps/blob/master/docs/CPP_STYLE.md)

### CODE REVIEW

File: docs/developer-guide/6-Contributing.md [hunk 1]
```diff
@@ -0,0 +1,245 @@
+# C++ Style Guide
+
+In general, [Google's coding standard](https://google.github.io/styleguide/cppguide.html) is used, and we strongly encourage to read it.
+
+Below are our specific (but not all!) exceptions to the Google's coding standard:
+
+- All C++ code should conform to the C++20 standard.
+- We use `.cpp` and `.hpp` files, not `.cc` and `.h` (`.c` and `.h` are used for C code), in UTF-8 encoding.
```
Comment (a-zakir) [inline docs/developer-guide/6-Contributing.md]: Actually we don’t use .hpp but .h

File: docs/developer-guide/6-Contributing.md [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+- File names are lowercase with underscores, like `file_reader.cpp`.
```
Comment (a-zakir) [inline docs/developer-guide/6-Contributing.md]: For future files?

File: docs/developer-guide/6-Contributing.md
Comment (pet-mit) [inline docs/developer-guide/6-Contributing.md:1]: we should unify this with "CONTRIBUTING.md" which is at the root of the repo

Comment (flomnes) [reply docs/developer-guide/6-Contributing.md:1]: done

**Review by pet-mit — APPROVED** [review]

**Pull request merged**